e2e/tests/stablex_test.rs

```diff
@@ -111,3 +120,124 @@ fn test_stablex() {
+
+    // Gather token and batch info
+    let token_a = instance
+        .token_id_to_address_map(0)
+        .call()
+        .wait()
+        .expect("Cannot get first Token address");
+    let token_b = instance
+        .token_id_to_address_map(7)
+        .call()
+        .wait()
+        .expect("Cannot get second Token address");
+    let batch = instance
+        .get_current_batch_id()
+        .call()
+        .wait()
+        .expect("Cannot get batchId");
+
+    // Approve Funds
+    let first_approve = IERC20::at(&web3, token_a)
+        .approve(instance.address(), 1_000_000.into())
+        .nonce(nonce)
+        .gas(1_000_000.into())
+        .gas_price(8_000_000_000u64.into())
+        .from(account.clone())
+        .send_and_confirm(Duration::from_secs(1), 1);
+    let second_approve = IERC20::at(&web3, token_b)
+        .approve(instance.address(), 1_000_000.into())
+        .nonce(nonce + 1)
+        .gas(1_000_000.into())
+        .gas_price(8_000_000_000u64.into())
+        .from(account.clone())
+        .send_and_confirm(Duration::from_secs(1), 1);
+
+    // Deposit Funds
+    let first_deposit = instance
+        .deposit(token_a, 1_000_000.into())
+        .nonce(nonce + 2)
+        .send_and_confirm(Duration::from_secs(1), 1);
+    let second_deposit = instance
+        .deposit(token_b, 1_000_000.into())
+        .nonce(nonce + 3)
+        .send_and_confirm(Duration::from_secs(1), 1);
+
+    // Place orders
+    let first_order = instance
+        .place_order(0, 7, batch + 2, 1_000_000.into(), 10_000_000.into())
+        .nonce(nonce + 4)
+        .send_and_confirm(Duration::from_secs(1), 1);
+    let second_order = instance
+        .place_order(7, 0, batch + 1, 1_000_000.into(), 10_000_000.into())
+        .nonce(nonce + 5)
+        .send_and_confirm(Duration::from_secs(1), 1);
+
+    // Wait for all transactions to be confirmed
+    println!("Waiting for transactions to be confirmed");
+    let results = join_all(vec![
+        first_approve,
+        second_approve,
+        first_deposit,
+        second_deposit,
+        first_order,
+        second_order,
+    ])
+    .wait();
+    for (index, result) in results.into_iter().enumerate() {
+        result.unwrap_or_else(|_| panic!("Tx #{} failed", index));
+    }
+
+    // Wait for solution to be applied
+    let sleep_time = instance
+        .get_seconds_remaining_in_batch()
+        .call()
+        .wait()
+        .expect("Cannot get seconds remaining in batch")
+        .low_u64()
+        + 30;
+
+    println!("Sleeping {} seconds...", sleep_time);
+    std::thread::sleep(Duration::from_secs(sleep_time));
+
+    // Make sure there was no error
+    let output = Command::new("docker-compose")
+        .arg("logs")
+        .output()
+        .expect("failed to execute process");
+    let logs = String::from_utf8(output.stdout).expect("failed to read logs");
+    assert!(!logs.to_lowercase().contains("error"));
```

Comment:
Super robust error checking :stuck_out_tongue:

Reply:
We print `docker-compose logs` on test failure, thus I'd expect the developer to identify what was the problem.